crates/tauri-cli/src/lib.rs

```diff
@@ -63,10 +63,17 @@ impl FromStr for ConfigValue {
     } else {
       let path = PathBuf::from(config);
       if path.exists() {
-        Ok(Self(serde_json::from_str(
-          &read_to_string(&path)
-            .with_context(|| format!("invalid configuration at file {config}"))?,
-        )?))
+        let raw = &read_to_string(&path)
+            .with_context(|| format!("invalid configuration at file {config}"))?;
+        if let Some(extension) = path.extension() {
+          match extension.to_str().expect("invalid unicode in file extension") {
+            "toml" => Ok(Self(::toml::from_str(raw)?)),
+            "json" | "json5" => Ok(Self(serde_json::from_str(raw)?)),
+            _ => anyhow::bail!("unsupported file type")
+          }
+        } else {
+          anyhow::bail!("invalid file path (requires file extension)")
+        }
```

Comment:
we can always default to JSON, and for Json5, we need to use `json5` crate
```suggestion
        match path.extension() {
          Some(ext) if ext == "toml" => Ok(Self(::toml::from_str(raw)?)),
          Some(ext) if ext == "json5" => Ok(Self(json5::from_str(raw)?)),
          // default to JSON
          _ => Ok(Self(serde_json::from_str(raw)?))
        }
```

Reply:
right!
isn't the usual behavior to try parsing with .json's with json5 as well?

Reply:
if a json5 file has comments in it, serde_json will not be able to parse it.

Reply:
I might have misunderstood your question, in that case yeah we could just use `json5` for `.json` files as well.

Reply:
you also need to update the documentation option to mention JSON5 support as well.

Reply:
> I might have misunderstood your question, in that case yeah we could just use `json5` for `.json` files as well.

From what i've seen in tauri-utils, we parse .json with serde, if we get an error we try with json5 and if json5 gives an error we return the serde error
I'll check again later tho

Reply:
yeah, I guess we can go that route as well